x509/ptr_sysptr_windows.go

```diff
@@ -0,0 +1,20 @@
+// Copyright 2018 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build go1.11
+
+package x509
+
+import (
+	"syscall"
+	"unsafe"
+)
+
+// For Go versions >= 1.11, the ExtraPolicyPara field in
+// syscall.CertChainPolicyPara is of type syscall.Pointer.  See:
+//   https://github.com/golang/go/commit/4869ec00e87ef
+
+func convertToPolicyParaType(p unsafe.Pointer) syscall.Pointer {
```

Comment:
Perhaps we can do something like this, hiding all the complexity inside this function here:
```go
func getCertChainPolicyPara(p *syscall.SSLExtraCertChainPolicyPara) *syscall.CertChainPolicyPara {
	return &syscall.CertChainPolicyPara{
		ExtraPolicyPara: (syscall.Pointer)(unsafe.Pointer(p))
}
```
and simplify the main code:

```go
para := getCertChainPolicyPara(sslPara)
```

Reply:
Simplifying the main code isn't necessarily something we want -- we (mostly) try to minimize deltas from the upstream codebase so that it's easier to merge in new versions.